src/views/Dissolutions/ReviewConfirmDissolution.vue

```diff
@@ -352,6 +409,39 @@ export default class ReviewConfirmDissolution extends Mixins(DateMixin) {
   get futureEffectiveDate (): string {
     return this.dateToPacificDateTime(this.getEffectiveDateTime.effectiveDate, true)
   }
+
+  /** Is true when the certify conditions are not met. */
+  private get isCertifyInvalid () {
+    return this.getValidateSteps && (!this.getCertifyState.certifiedBy || !this.getCertifyState.valid)
+  }
+
+  /** Handler for Valid change event. */
+  private onIsCertified (val: boolean): void {
+    this.setCertifyState(
+      {
+        valid: val,
+        certifiedBy: this.getCertifyState.certifiedBy
+      }
+    )
+  }
+
+  /** Handler for CertifiedBy change event. */
+  private onCertifiedBy (val: string): void {
+    this.setCertifyState(
+      {
+        valid: this.getCertifyState.valid,
+        certifiedBy: val
+      }
+    )
+  }
+
+  private onOptionalEmailChanged (optionalEmail: string): void {
```

Comment:
Since this is a trivial method (and also line 442), you can put it right in the component declaration in the template.

Up to you. (IMO, on the one hand, it means the template calls store methods, which some people might think is bad form (ie, should strive for clean interfaces re: MVVM). OTOH, the template and script for this component is basically "one combined thing" so this is actually cleaning up the code somewhat. If it was me, I'd remove the unnecessary method in the script. But I approve this code either way.)

Reply:
(By the way, some links on MVVM:
https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel
https://trio.dev/blog/vue-vs-angular
https://subscription.packtpub.com/book/web-development/9781786469946/2/ch02lvl1sec18/mvvm-architectural-pattern

Due to .vue files containing both the template and the script, the interface between view/viewmodel/model is blurred or non-existent. However, it still makes for a cleaner project -- easier to understand and extend -- when we think about these interfaces as we implement.

I just want you guys to know why I keep making comments about "architecture" 😀 But we're in good shape. Thanks all for your efforts!)